apps/common-app/src/examples/RuntimeTests/tests/animations/withDelay/keepSnapshot.test.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useCallback, useEffect } from 'react';
+import { View, StyleSheet } from 'react-native';
+import Animated, {
+  useSharedValue,
+  useAnimatedStyle,
+  withTiming,
+  withDelay,
+  withSpring,
+  withSequence,
+} from 'react-native-reanimated';
+import {
+  describe,
+  test,
+  expect,
+  render,
+  useTestRef,
+  getTestComponent,
+  wait,
+  recordAnimationUpdates,
+  unmockAnimationTimer,
+  mockAnimationTimer,
+  clearRenderOutput,
+} from '../../../ReanimatedRuntimeTestsRunner/RuntimeTestsApi';
+import type { SingleViewSnapshot } from '../../../ReanimatedRuntimeTestsRunner/matchers/snapshotMatchers';
+
+enum TestAnimation {
+  TIMING = 'withTiming',
+  SEQUENCE = 'sequence of withTiming and withSpring',
+}
+
+const TEST_COMPONENT_ACTIVE_REF = 'TestComponentActive';
+const TEST_COMPONENT_PASSIVE_REF = 'TestComponentPassive';
+const TestComponent = ({
+  applyWithDelay,
+  testAnimation,
+  delay,
+}: {
+  applyWithDelay: boolean;
+  testAnimation: TestAnimation;
+  delay: number;
+}) => {
+  const widthActiveSV = useSharedValue(100);
+  const widthPassiveSV = useSharedValue(100);
+
+  const refActive = useTestRef(TEST_COMPONENT_ACTIVE_REF);
+  const refPassive = useTestRef(TEST_COMPONENT_PASSIVE_REF);
+
+  const stylePassive = useAnimatedStyle(() => {
+    return {
+      width: widthPassiveSV.value,
+    };
+  });
+
+  const getBaseAnimation = useCallback(
```

Comment:
Do we need `useCallback` here? 🤔

Reply:
Yes, it is listed among dependencies of `useEffect` in line 81.